diff --git a/node_modules/@react-pdf/layout/lib/index.js b/node_modules/@react-pdf/layout/lib/index.js
index b7ea5e3..2838345 100644
--- a/node_modules/@react-pdf/layout/lib/index.js
+++ b/node_modules/@react-pdf/layout/lib/index.js
@@ -3039,10 +3039,12 @@ const paginate = (page, pageNumber, fontStore, yoga) => {
         return [];
     if (page.props?.wrap === false)
         return [page];
+    dispatchEvent(new Event('layoutPage'));
     let splittedPage = splitPage(page, pageNumber, fontStore, yoga);
     const pages = [splittedPage[0]];
     let nextPage = splittedPage[1];
     while (nextPage !== null) {
+        dispatchEvent(new Event('layoutPage'));
         splittedPage = splitPage(nextPage, pageNumber + pages.length, fontStore, yoga);
         pages.push(splittedPage[0]);
         nextPage = splittedPage[1];
