diff --git a/node_modules/png-js/lib/png-js.browser.js b/node_modules/png-js/lib/png-js.browser.js
index 4c04414..eeb92f7 100644
--- a/node_modules/png-js/lib/png-js.browser.js
+++ b/node_modules/png-js/lib/png-js.browser.js
@@ -1,6 +1,7 @@
-import { unzlib } from 'fflate';
-
-const inflate = unzlib ;
+import { unzlibSync } from 'fflate';
+const inflate = (data, cb) => {
+  cb(null, unzlibSync(data));
+};
 
 class PNG {
   static decode(path, fn) {
